src/test/java/com/googleinterns/zoomtube/servlets/DiscussionServletTest.java

```diff
@@ -0,0 +1,205 @@
+    assertThat(comments.isEmpty()).isTrue();
+  }
+
+  @Test
+  public void doGet_returnsComments() throws ServletException, IOException {
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+    datastore.put(createTestCommentEntity(LECTURE_ID));
+    datastore.put(createTestCommentEntity(LECTURE_ID));
+    datastore.put(createTestCommentEntity(LECTURE_ID));
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json;");
+    List<Comment> comments = getCommentsFromJson(content.toString());
+    assertThat(comments.size()).isEqualTo(3);
+  }
+
+  @Test
+  public void doGet_returnsCommentsForSpecificLecture() throws ServletException, IOException {
+    final int lectureA = 1;
+    final int lectureB = 2;
+    final int lectureC = 3;
+    // Looking for two comments under LECTURE_A.
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE))
+        .thenReturn(Integer.toString(lectureA));
+    datastore.put(createTestCommentEntity(lectureB));
+    datastore.put(createTestCommentEntity(lectureC));
+    datastore.put(createTestCommentEntity(lectureA)); // Add the first.
+    datastore.put(createTestCommentEntity(lectureB));
+    datastore.put(createTestCommentEntity(lectureB));
+    datastore.put(createTestCommentEntity(lectureC));
+    datastore.put(createTestCommentEntity(lectureA)); // Add the second.
+    datastore.put(createTestCommentEntity(lectureC));
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json;");
+    List<Comment> comments = getCommentsFromJson(content.toString());
+    assertThat(comments.size()).isEqualTo(2);
+  }
+
+  private Entity createTestCommentEntity(int lectureId) {
+    Entity commentEntity = new Entity(Comment.ENTITY_KIND);
+    Key lecture = KeyFactory.createKey(/* kind= */ "Lecture", lectureId);
+    commentEntity.setProperty(Comment.PROP_LECTURE, lecture);
+    commentEntity.setProperty(Comment.PROP_PARENT, null);
+    commentEntity.setProperty(Comment.PROP_TIMESTAMP, new Date(0));
+    commentEntity.setProperty(Comment.PROP_AUTHOR, new User(DEFAULT_EMAIL, AUTH_DOMAIN));
+    commentEntity.setProperty(Comment.PROP_CONTENT, TEST_COMMENT_CONTENT);
+    commentEntity.setProperty(Comment.PROP_CREATED, new Date(Clock.systemUTC().millis()));
+
+    return commentEntity;
+  }
+
+  private List<Comment> getCommentsFromJson(String json) {
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    Type listType = new TypeToken<ArrayList<Comment>>() {}.getType();
+    List<Comment> comments = gson.fromJson(json, listType);
```

Comment:
can you do: return gson.fromJson(json, listType);

Reply:
changed